### ISSUE TITLE
In-Game: [YYC] Crash with "HASH argument is unset" when repeatedly calling with() and variable_clone() on an empty constructor instance

### ISSUE THREAD

**Issue by JonathanHackerCG** [YoYoGames/GameMaker-Bugs#5923]
### Description
The following code crashes only on YYC (Windows) with the error message "HASH argument is unset". Behaves as expected on VM.

```
function Foo() constructor
{
	//Empty
}

a = new Foo();
with (a)
{
	//Do nothing.
}

b = variable_clone(a);
with (b)
{
	//Do nothing.
}

c = variable_clone(b); //CRASH "HASH argument is unset"
```

I only tested on v2024.4.1, the issue may exist in older versions since I haven't built for YYC in a few months due to other issues that have since been fixed.

### Steps To Reproduce
Run the included code, or the attached project.

### Which version of GameMaker are you reporting this issue for?
IDE v2024.4.1.152 Runtime v2024.4.1.201

### Which operating system(s) are you seeing the problem on?
Windows 10.0.19045.0

### Which platform(s) are you seeing the problem on?
Windows

[9acf2358-77c0-485c-a4a6-572774582b70](https://api.gamemaker.io/api/github/downloads/9acf2358-77c0-485c-a4a6-572774582b70)

**Comment by tinkerer-red** [comment]
Also broken on `IDE v2024.2.0.132  Runtime v2024.2.0.163`

**Comment by JonathanHackerCG** [comment]
Still happening in ``IDE v2024.6.2.162 Runtime v2024.6.1.208``. Additional investigation in the [GM Kitchen Discord](https://discord.com/channels/724320164371497020/1009467299956269076/1263514230682816586) shows that specifying the context also does not crash, like so. (global in a script, id in an instance Create event).

```foo = function() constructor {
    /* empty */
}

a = new foo();
with (a) {
    /* empty */
}

b = variable_clone(id.a)
with (b) {
    /* empty */
}

c = variable_clone(id.b);
```

**Comment by DiasFranciscoA** [comment]
[INTERNAL] @rwkay This is a problem in the asset compiler's generated code (the scope is being handled incorrectly)

**Comment by rwkay** [comment]
Fixed in 2024.10

**Comment by mgeddesGM** [comment]
verified as of  IDE v2024.1100.0.626  Runtime v2024.1100.0.652
